### DISCUSSION TITLE
403 errors with some thumbnails Some load fine?

### DISCUSSION THREAD

**Discussion by gnossosp** [openstore-ecommerce/OpenStore#266] [Q&A] [answered]
DNN Version v. 09.10.02 (0)
.NET FRAMEWORK 4.8
Store Version   v04.01.06
Site opens slowly sometimes, showing modal loading window/icon. Upon loading certain random thumbnails appear, while some are (again apparently at random) missing with a 403 error. 
Permissions are set to read for the default app pool. As a test I have changed to full control with same errors resulting. 
Testing in
Firefox 99.0 (mx linux)
Chrome Version 100.0.4896.127 (Win 10 pro 21H2  19044.1645 )
Brave V1.37.116 (Apr 14, 2022) Upgraded Chromium to 100.0.4896.127
Firefox 99.0.1
Microsnot Edge Version 100.0.1185.44 

Any ideas as to solution to this problem? Thanks

![openstore](https://user-images.githubusercontent.com/72934916/163854682-643bfa57-e50f-4498-ac02-f23593511e68.jpg)

**Comment by DNNMonster** [comment, marked as answer]
I think you're right investigating the permissions.  I cannot think of another reason why you'd get the 403.  I would try again setting the permissions to full control for the site's app pool on your www root directory since DNN and OS both require it.  I am not 100% sure if you have to, but I would also restart IIS and/or even bounce the server if possible.  After that I would double check the Portal's directory to make sure the permissions are getting applied.  Also, you might verify that the NBStore directories in your Portal are also having the proper permissions applied.  Lastly, you could log in as the OS admin and also try validating the store and purging the unused images though I doubt this is responsible for the 403.  Things to try but not a guaranteed solution. 

You can report back if none of that helps and we'll try to sleuth it out.

**Reply by gnossosp** [reply]
- Triple checked app pool permissions
- App pool permissions are propagating to all directories
- Refreshed/Restarted app pool for IIS
- and disabled (this might not be it but - why not) "Preventing DoS attacks by IIS dynamic IP restriction" in IIS settings
I searched and could not find any meaningful discussion about changing the default values in the "Preventing DoS.." module.

Today I have cleared browsers on two machines and can not reproduce the error. So I don't know what the answer is/was.... marking this closed. Thanks

**Reply by leedavi** [reply]
We had this kind of problem on our website (not OpenStore) if we dis-activated browser cache.

No idea why, all the settings looked good....  It was  the same for Edge and Chrome, but in FF it worked.

In the end we moved the website to another server and it was everything started working.

Very strange and still don't know why.  I'm guessing a problem with the browser.
